src/miniflask/miniflask.py

```diff
@@ -470,7 +470,7 @@ def _settings_parser_add(self, varname, val, caller_traceback, nargs=None, defau
         if argtype == Enum:
             kwarg["action"] = EnumAction
             kwarg["type"] = val if isinstance(val, EnumMeta) else type(val)
-        elif argtype == str2bool and nargs != '+':  # pylint: disable=comparison-with-callable
+        elif argtype == str2bool and nargs != '*':  # pylint: disable=comparison-with-callable
```

Comment:
Exactly, in f0fc206, I have enabled lists to be empty as well.
With that commit I introduced the bug, that disabled lists for boolean values, because I did not adapt that line here...
(The new test cases should catch these kinds of mistakes.)